packages/discord.js/src/client/Client.js

```diff
@@ -230,6 +230,8 @@ class Client extends BaseClient {
       }
 
       this.options.ws.presence = this.presence._parse(this.options.presence);
+    } else if (this.options.ws.presence) {
+      this.options.ws.presence = this.presence._parse(this.options.ws.presence);
```

Comment:
This should parse `this.options.ws.initialPresence` with the ws.presence as fallback with nullish coalesce.